### PR TITLE
Add schedule feature and associated tests

### PR DESCRIPTION
**Pull request by therealfrankshaw**
Adding new schedule feature to gem. Allows users to grab NBA and MLB future game times. It seems ESPN is converting other major sports to same format, so adding NFL, NHL, NCB, and NCF, will be pretty simple in the future.

### CODE REVIEW

**Comment by aj0strow** [comment]
Thanks for this, will look into it on the weekend.

**Comment by aj0strow** [comment]
@BjjMeche this is great, schedules should be part of this gem. 

Would you mind refactoring the `__callee__`? I've never seen it before, don't see the point. It seems like it would be more DRY to define in a loop. 

```ruby
%w(nba mlb).each do |league|
  define_method ...
end
```

File: lib/espn_scraper.rb [hunk 1]
```diff
@@ -1,3 +1,3 @@
-%w[ boilerplate teams scores ].each do |file|
-  require "espn_scraper/#{ file }"
-end
\ No newline at end of file
+%w[ boilerplate teams scores schedules ].map do |file|
```
Comment (BjjMeche) [inline lib/espn_scraper.rb]: This doesn't seem to work without having line 2 be require_relative

File: lib/espn_scraper/schedules.rb [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    # {
+    #  :away_team_name=>"Sacramento Kings",
+    #  :away_data_name=>"sac",
+    #  :home_team_name=>"Denver Nuggets",
+    #  :home_data_name=>"den",
+    #  :match_time=>"21:00"
+    #  }
+
+    def get_nba_schedules(date)
+      Schedules.get_schedule("nba", date)
```
Comment (BjjMeche) [inline lib/espn_scraper/schedules.rb]: changed per req

**Comment by aj0strow** [comment]
Why are past dates not allowed? Seems like they are: http://espn.go.com/mlb/schedule?date=20150430

**Comment by austinarchibald** [comment]
What's the latest on this? Having the scheduling feature is exactly what I am looking for. I'm hoping to use this asap. Thank you!

**Comment by aj0strow** [comment]
@austinarchibald yeah schedules would be sweet

@BjjMeche the only reason i haven't merged is i was mostly wondering about this choice:

```ruby
+       if date <= Date.today
+          puts "ERROR: Digging Into Past Games Not Allowed"
+          false
```

It's not accurate to ignore past schedules because they do exist on ESPN. It also makes unit testing impossible because there's no way to know what schedules will be in the future so you can't verify the parsing is getting the correct values.

**Comment by austinarchibald** [comment]
Hey @BjjMeche, I tried your repo just now, and it works great (I can pull schedule), but then it breaks the get_xxx_scores. e.g. line 1 returns an emtpy array. When I switch back to @aj0strow repo, scores works.

```ruby
<%= ESPN.get_mlb_scores( Date.parse('Jul 5, 2015') ) %>  # => []
<%= ESPN.get_mlb_schedules( Date.parse('Jul 7, 2015') ) %>  # => [{:away_team_name=>"St. Louis Cardinals", :away_data_name=>"stl", :home_team_name=>"Chicago Cubs", :home_data_name=>"chc", :match_time=>"17:20"}, etc ]
```

If I can just get the scores to work on your repo, then I'm good to go. I can set up some conditions to handle whether it uses scores or schedules based on today's date. @BjjMeche - Would you mind taking a look at it? Much appreciated, thank you! This is awesome!

**Comment by BjjMeche** [comment]
Hey there guys! for some reason my notifications are turned off for this branch. I'm not purposely ignoring you!

So i think the reason I made the schedule feature future searchable only was because I was not anticipating anyone searching for games in the past. I would have to add additional logic to scrape that data, but it would not be impossible. 

I'll make a PR by midnight on Sunday adding these features. Looks like NFL made the format switch to match MLB and NBA. Adding 1 method will give us the same features. Awesome.

**Comment by dgmdan** [comment]
I'd like to close this if the feature is no longer being worked on.

**Pull request closed without merging**